Clients/GUI/Pages/UserInfo.razor

```diff
@@ -3,162 +3,295 @@
+                       class="form-control col-3 button btn-primary" value="@(IsPasswordBeingChanged ? "Cancel changing password" : "Change password")" />
+                <br />
+                <input type="@(IsAvatarBeingChanged || IsPasswordBeingChanged || IsUserInfoBeingChanged ? "button" : "hidden")"
+                       @onclick="@DiscardAllChanges" class="form-control col-3 button btn-primary" value="Discard all changes" />
+                <br />
+                <ValidationSummary />
+                <h4 style="display:@(ErrorOccured ? "block" : "hidden"); color:red">@ErrorMessage</h4>
+                <input type="button" @onclick="@EditInfoClickHandler" class="form-control col-3 button btn-primary" value="@(IsEditModeDisabled ? "Edit" : "Save changes")" />
+            </EditForm>
         </div>
-        <div class="form-group" style="@(IsPasswordBeingChanged ? "" : "height: 0; color: transparent")">
-            <label for="NewPassword">Enter new password: </label>
-            <InputText type="@(IsPasswordBeingChanged ? "password" : "hidden")" class="@InputStyleClass" disabled="@(!IsPasswordBeingChanged)" @bind-Value="@user.NewPassword" />
+    </div>
         </div>
-        <br />
-        <input type="@(IsEditModeDisabled ? "hidden" : "button")" @onclick="@ChangePasswordClickHandler" class="form-control col-3 button btn-primary" value="@(IsPasswordBeingChanged ? "Cancel changing password" : "Change password")" />
-        <br />
-        <ValidationSummary />
-        <h4 style="display:@(ErrorOccured ? "block" : "hidden"); color:red">@ErrorMessage</h4>
-        <input type="button" @onclick="@EditInfoClickHandler" class="form-control col-3 button btn-primary" value="@(IsEditModeDisabled ? "Edit" : "Save changes")" />
-    </EditForm>
-</div>
-
-@code {
-    public bool IsEditModeDisabled { get; set; }
-    public bool IsPasswordBeingChanged { get; set; }
-    public bool PopupVisible { get; set; }
-    public string ErrorMessage { get; set; }
-    public string InputStyleClass { get; set; }
-    public bool ErrorOccured { get; set; }
-    private UserInfoViewModel user;
-    private UserToken userToken;
-
-    private async Task EditInfoClickHandler()
-    {
-        if (IsEditModeDisabled)
-        {
-            IsEditModeDisabled = false;
-            InputStyleClass = "form-control col-3";
-        }
-        else
-        {
-            try
+
+        @code {
+
+            private bool IsEditModeDisabled { get; set; }
+            private bool IsUserInfoBeingChanged { get; set; }
+            private bool IsPasswordBeingChanged { get; set; }
+            private bool IsAvatarBeingChanged { get; set; }
+            private bool PopupVisible { get; set; }
+            private string ErrorMessage { get; set; }
+            private string InputStyleClass { get; set; }
+            private bool ErrorOccured { get; set; }
+            private UserInfoViewModel user;
+            private UserInfoViewModel oldUser;
```

Comment:
Wait, why do we need 2 view models?

Reply:
To have the opportunity to discard changes

Reply:
> To have the opportunity to discard changes

Then why can we discard changes only while changing password?

Reply:
Nope. We can discard changes every time when we change avatar, password or click on the input text (name , lastname, birthday)